test/unit/file_transfer_agent/file_transfer_agent_test.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+const assert = require('assert');
+const testUtil = require('../../integration/testUtil');
+const os = require('os');
+const fs = require('fs');
+const crypto = require('crypto');
+const getMatchingFilePaths = require('./../../../lib/file_transfer_agent/file_transfer_agent').getMatchingFilePaths;
+
+
+function clean(randomName) {
+  const matchedFiles = getMatchingFilePaths(os.tmpdir(), `${randomName}matched` + '*', 'win32');
+  const notmatchedFiles = getMatchingFilePaths(os.tmpdir(), `${randomName}notmatched` + '*', 'win32');
+
+  matchedFiles.forEach(filePath => {
+    fs.rmSync(filePath);
+  });
+  notmatchedFiles.forEach(filePath => {
+    fs.rmSync(filePath);
+  });
+}
+
+function createFiles(randomName, options) {
+  for (let i = 0; i < 3; i++) {
```

Comment:
can we move 3 to a paramater? or make it constant? to avoid magic numbers in test assertions